### ISSUE TITLE
[tests] Replace usage of org.eclipse.xtext.junit4 by org.eclipse.xtext.testing

### ISSUE THREAD

**Issue by kthoms** [eclipse/xtext-xtend#29]
The Xtend tests are still using the deprecated org.eclipse.xtext.junit4 bundle, they should use the org.eclipse.xtext.testing bundle now.

**Comment by kthoms** [comment]
<h1>Replacements</h1>
<table>
<tr><th>old class</th><th>new class</th></tr>
<tr><td>org.eclipse.xtext.junit4.InjectWith</td><td>org.eclipse.xtext.testing.InjectWith</td></tr>
<tr><td>org.eclipse.xtext.junit4.XtextRunner</td><td>org.eclipse.xtext.testing.XtextRunner</td></tr>
<tr><td>org.eclipse.xtext.junit4.logging.LoggingTester</td><td>org.eclipse.xtext.testing.logging.LoggingTester</td></tr>
<tr><td>org.eclipse.xtext.junit4.util.ResourceHelper</td><td>org.eclipse.xtext.testing.util.ResourceHelper</td></tr>
<tr><td>org.eclipse.xtext.junit4.validation.ValidationTestHelper</td><td>org.eclipse.xtext.testing.validation.ValidationTestHelper</td></tr>
<tr><td>org.eclipse.xtext.junit4.internal.LineDelimiters</td><td>org.eclipse.xtext.xbase.junit.LineDelimiters</td></tr>
<tr><td>org.eclipse.xtext.junit4.TemporaryFolder</td><td>org.eclipse.xtext.xbase.testing.TemporaryFolder</td></tr>
<tr><td>org.eclipse.xtext.junit4.formatter.FormatterTester</td><td>org.eclipse.xtext.testing.formatter.FormatterTestHelper</td></tr>

</table>

<h1>Missing framework classes</h1>
Some classes used from xtext.junit4 are not available in xtext.testing. I assume they should be copied.
- org.eclipse.xtext.junit4.util.InMemoryURIConverter
- org.eclipse.xtext.junit4.internal.StopwatchRule
- org.eclipse.xtext.junit4.internal.Timed

Until these classes are available, the tests have to rely on both bundles.

org.eclipse.xtend.core.tests.typesystem.ErrorTest.ResolveBatchedWithoutJRE extends org.eclipse.xtext.xbase.junit.typesystem.NoJRESmokeTester, which itself depends on xtext.junit4. This has also to be refactored in a separate issue.

**Comment by meysholdt** [comment]
@kthoms could you comment on how much of this issue is left after #30 got merged?

**Comment by cdietrich** [comment]
at least the old dependency should be removed

**Comment by cdietrich** [comment]
i am preparing a bunch more of adaptations

**Comment by cdietrich** [comment]
see https://github.com/eclipse/xtext-xtend/pull/31 for that other bunch

**Comment by spoenemann** [comment]
Finishing this would allow to fix #58.

**Comment by cdietrich** [comment]
@kthoms @spoenemann do we have a list of left-overs somewhere?

**Comment by cdietrich** [comment]
i asume we have no leftovers now. correct?

**Comment by cdietrich** [comment]
got no objects so i will close this now
